src/pkg/services/m365/api/mail_test.go

```diff
@@ -411,3 +397,112 @@ func (suite *MailAPIIntgSuite) TestRestoreLargeAttachment() {
 	require.NoError(t, err, clues.ToCore(err))
 	require.NotEmpty(t, id, "empty id for large attachment")
 }
+
+func (suite *MailAPIIntgSuite) TestMail_GetContainerByName() {
```

Comment:
please add a subfolder to `mail_get_container_by_name` which has the same name as the parent and add a test case to fetch the child folder by name

Reply:
done